index.js

```diff
@@ -118,8 +118,95 @@ async function getAllDatasets(dataCatalogUrl = 'https://openactive.io/data-catal
   return { jsonld: jsonldFromDatasetUrls, errors };
 }
 
+/**
+ * Validates JSON-LD content by ensuring the '@id' or 'id' field matches the provided ID.
+ *
+ * This function performs an HTTP GET request to the specified ID (URL), retrieves
+ * the response, and extracts JSON-LD from it if needed and possible. It then compares
+ * the '@id' or 'id' field from the retrieved JSON-LD to the provided ID.
+ *
+ * Note that this is only applicable to JSON-LD "@id" for the DataCatalog and Dataset types, which must resolve.
+ *
+ * @async
+ * @function validateJsonLdId
```

Comment:
(just a question :) ). Why do you include these? VSCode (at least) can already determine these just from the function signature

Reply:
Great question: AI made me do it :) will remove